tests/PHPStan/Analyser/data/math.php

```diff
@@ -109,4 +109,26 @@ public function doSit(int $i, int $j): void
 		assertType('int', $i - $j);
 	}
 
+	/**
+	 * @param int<-5, 5> $range
+	 */
+	public function multiplyZero(int $i, float $f, $range): void
+	{
+		assertType('0', $i * 0);
+		assertType('0', $f * 0);
```

Comment:
Float multiplied by zero is a float: https://3v4l.org/u6dA6

It also works for strings: https://3v4l.org/Ksheh

Reply:
good catch. fixed